### PR TITLE
Debug.log format in Elchemy instead of Elixir

### PR DESCRIPTION
**Pull request by latachz**

### CODE REVIEW

File: lib/format.ex [hunk 1]
```diff
@@ -25,7 +22,8 @@ defmodule Elchemy.Format do
       if is_atom(head) do
         false
       else
-        term |> List.to_string
+        term = [head, rest]
+        List.to_string(term)
```
Comment (wende) [inline lib/format.ex]: I think you lost a pipe (|>)

File: lib/format.ex [hunk 2]
```diff
@@ -15,7 +15,12 @@ defmodule Elchemy.Format do
   def inspect(term) when is_tuple(term) do
       [head | _rest] = term |> Tuple.to_list
       if is_atom(head) do
-          term |> Tuple.to_list() |> Enum.map(&inspect/1) |> Enum.join(" ") |> String.capitalize
+        inner = term |> Tuple.to_list() |> Enum.map(&inspect/1) |> Enum.join(" ")
+        if(String.length() == 1) do ###JAK TO ZROBIC?
```
Comment (OvermindDL1) [inline lib/format.ex]: Uh, `String.length/0` doesn't exist?  o.O

File: lib/format.ex [hunk 3]
```diff
@@ -15,7 +15,12 @@ defmodule Elchemy.Format do
...
+          term |> String.upcase()
```
Comment (OvermindDL1) [inline lib/format.ex]: Why have this branch at all, why not just capitalize regardless as if it is a single character then it should(?) get capitalized anyway?

Comment (latachz) [reply lib/format.ex]: with String.capitalize()
```
left:  "A 1 b 2 3 4"
right: "A 1 (B 2 3) 4"
```
with String.upcase()
```
left:  "A 1 b 2 3 4"
right: "A 1 (B 2 3) 4"
```

Comment (OvermindDL1) [reply lib/format.ex]: That seems to be precisely what I expect from them though?

Also it doesn't matter as the first branch is only taken when the string has a length of 1 (assuming `term` is supposed to be in it), so it wouldn't matter anyway?

File: lib/format.ex [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  @doc "Some doc"
+  @spec inspect(term()) :: String.t()
+  def inspect(term, in_type \\ false)
+  def inspect(term, _) when is_map(term) do
+    case Map.to_list(term) do
+      [] -> "Dict.fromList []"
+    [{key, _value} | _] when is_atom(key) ->
+      inner = term |> Enum.map(fn {key, value} ->
+        (key |> to_string()) <> " = " <> (value |> inspect())
```
Comment (wende) [inline lib/format.ex]: `to_string(key)` looks better

Comment (wende) [reply lib/format.ex]: same for `inspect(value)`

File: lib/format.ex [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      end)
+      "{ " <> Enum.join(inner, ", ") <> " }"
+    [{_key, _value} | _] ->
+      inner = term |> Enum.map(fn {key, value} ->
+        (inspect(key)) <> " = " <> (value |> inspect())
```
Comment (wende) [inline lib/format.ex]: Same as above + unnecessary parenthesis

File: lib/format.ex [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      else
+        inner = term |> Tuple.to_list() |> Enum.map(&inspect/1)
+        "(" <> Enum.join(inner, ", ") <> ")"
+      end
+  end
+
+  def inspect(term, _) when is_list(term) do
+    inner =
+      term
+      |> Enum.map(fn x -> inspect(x) end)
```
Comment (wende) [inline lib/format.ex]: Could use `&inspect/1` instead

File: lib/format.ex [hunk 7]
```diff
@@ -2,9 +2,14 @@ defmodule Elchemy.Format do
...
 
-  @doc "Some doc"
+  @moduledoc """
+    Formating Elixir to Elm message type.
+  """
+
   @spec inspect(term()) :: String.t()
   def inspect(term, in_type \\ false)
+
+  @doc "When term is map"
```
Comment (wende) [inline lib/format.ex]: A doc is only considered once for all of the function heads, so instead it should be a description of what the function does, and be located only at the top above of the first head definition (line 10)

File: lib/format.ex [hunk 8]
```diff
@@ -2,9 +2,14 @@ defmodule Elchemy.Format do
 
   import Kernel, except: [inspect: 1, inspect: 2]
 
-  @doc "Some doc"
+  @moduledoc """
+    Formating Elixir to Elm message type.
```
Comment (wende) [inline lib/format.ex]: > Module for inspecting a string representation of data structures according to Elchemy syntax instead of Elixir syntax

Comment (wende) [reply lib/format.ex]: That'd fit the description a bit better

File: lib/format.ex [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+    if is_atom(head) do
+      type = head |> Atom.to_string() |> String.capitalize()
+      args = rest |> Enum.map(&inspect(&1, true)) |> Enum.join(" ")
+
+      if in_type,
+        do: ["(", type, " ", args, ")"],
+        else:
+          [type, " ", args]
+          |> Enum.join()
```
Comment (wende) [inline lib/format.ex]: This looks misleading

**Pull request merged**